File: PRACTICAS/practica1/Civitas_Ruby/lib/test_p1.rb
require './Operaciones_juego'
require './casilla'
require './civitas'
require './diario'
require './estados_juego'
require './mazo_sorpresas'
require './sorpresa'
require './tablero'
require './tipo_casilla'
require './tipo_sorpresa'
require './dado'

module Civitas
  


class TestP1
  
  def self.main
    
    
    
    #Punto 1
    
    c0=0
    c1=0
    c2=0
    c3=0
    
    #dado = Dado.send :new
    
    for i in(0..99)
      
      num = Dado.instance.quienempieza(4)
      
      if(num==0)
        
        c0 += 1
        
      end
        
      if (num==1)
          
          c1 += 1
      end
      
      
      if (num==2)
          
          c2 += 1
      end  
      
      
      if (num==3)
          
          c3 += 1
            
      end
    
    
    end
    
    puts c0
    puts c1
    puts c2
    puts c3
    
    #Punto 2
    
    for i in (0..40)
      
      if(i < 20)
       
        tirada = Dado.instance.tirar
        puts tirada
      else
       
        #Dado.instance.setdebug(true)     lo quito para probar el ocurre evento
        tirada = Dado.instance.tirar
        puts tirada
        
      end
     
    end
      
   #Punto 3
   
    res = Dado.instance.getultimoresultado
    puts "Resultado final: #{res}"
    
    
    
    if(Dado.instance.salgodelacarcel)
      puts "puedes salir de la carcel"
    else
      puts "no puedes salir de la carcel"
    end
    
    
    #Punto 4
    
    puts TipoCasilla::CALLE
    puts TipoSorpresa::IRCARCEL
    puts Estados_juego::DESPUES_CARCEL
    
    #Punto 5
    
    valor = true
    mazo = MazoSorpresas.new(valor)
    mazo.almazo("nueva")
    mazo.almazo("otra")
    mazo.siguiente
    mazo.inhabilitarcartaespecial("otra")
    mazo.habilitarcartaespecial("otra")
    puts Diario.instance.leer_evento
    puts Diario.instance.leer_evento
    
    pendiente = Diario.instance.eventos_pendientes
    puts pendiente
    
    #Punto 6
    
    
    
    
    #Punto 7

    tab = Tablero.new(4)
    
    for i in (0..12)
      
      if i < 5
        
        s = Casilla.new("salida")
        tab.aniadecasilla(s)
        
      end
      
      if i == 5
        
        j = Casilla.new("carcel")
        tab.aniadecasilla(j)
        
      end
      
      if i == 6
        
        k = Casilla.new("descanso")
        tab.aniadecasilla(k)
        
      end
      
      if i == 7
        
        l = Casilla.new("impuesto")
        tab.aniadecasilla(l)
        
      end
      
      if i == 8
        
        m = Casilla.new("calle")
        tab.aniadecasilla(m)
        
      end
      
      Dado.instance.setdebug(true)
      
      for i in (0..12)
        
        Dado.instance.tirar
        ul = Dado.instance.getultimoresultado
        puts ul 
        tab.getcasilla(numCasilla)
        
      end
      
      
    end
    
    
    
  end
    
  main  
    
  end
  
  
  end
  

File: PRACTICAS/practica1/Civitas_P1/src/civitas_p1/Test_P1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

/**
 *
 * @author xaviv
 */
public class Test_P1 {
    public static void main(String[] args) {
        
        
        int contador0, contador1, contador2, contador3, x;
        
        
        contador0 = contador1 = contador2 = contador3 = 0;
          
        for (int i = 0; i < 100; i++){
            
            x = Dado.getInstance().quienEmpieza(4);
            
            if (x == 0){
                contador0 ++;
            }
            if (x == 1){
                contador1 ++;
            }
            if (x == 2){
                contador2 ++;
            }
            if (x == 3){
                contador3 ++;
            }
            
        }
      
        System.out.println("\nContador 0: " + contador0 + "\nContador 1: " + contador1 + "\nContador 2: " + contador2 + "\nContador 3: " + contador3);
     
        
        for (int i = 0; i < 20; i++){
            
            if (Dado.getInstance().salgoDeLaCarcel())
                System.out.println( "\n" + Dado.getInstance().getUltimoResultado());
            else
                System.out.println( "\nNo salgo");
            
            
                
        }
        
        
        MazoSorpresas españa = new MazoSorpresas();
        Sorpresa s1 = new Sorpresa(TipoSorpresa.PAGARCOBRAR, ), s2 = new Sorpresa(TipoSorpresa.PORJUGADOR);
        
        españa.alMazo(s1);
        españa.alMazo(s2);
        
        System.out.println( "\n" + españa.getSorpresa(0).getTipo() + "\n" + españa.getSorpresa(1).getTipo());
        
        españa.inhabilitarCartaEspecial(s2);
        
        españa.habilitarCartaEspecial(s2);
        
        System.out.println( "\n" + Diario.getInstance().leerEvento() + "\n" + Diario.getInstance().leerEvento());
        
        if (Diario.getInstance().eventosPendientes())
            System.out.println("\n ohhhh");
        else
            System.out.println("\n ahhhh");
        
        
        Tablero monotu = new Tablero(6);
        
        for(int i = 0; i < 12; i++){
            
            if (i < 5){
                Casilla s = new Casilla("Calle", TipoCasilla.CALLE);
                monotu.añadeCasilla(s);
            }
            else if (i == 5){
                Casilla s = new Casilla("Sorpresa", TipoCasilla.SORPRESA);
                monotu.añadeCasilla(s);
            }
            else if (i == 6){
                Casilla s = new Casilla("Aparcamiento", TipoCasilla.DESCANSO);
                monotu.añadeCasilla(s);
            }
            else if (i == 7){
                monotu.añadeJuez();
            }
            else if (i == 8){
                Casilla s = new Casilla("Paga", TipoCasilla.IMPUESTO);
                monotu.añadeCasilla(s);
            }
            else{
                Casilla s = new Casilla("Calle", TipoCasilla.CALLE);
                monotu.añadeCasilla(s);
            }
        }
        
        
        int cont = 0, pos_actual = 0;
       
        
        
        while (cont < monotu.getSize()){
            
            System.out.println("\n" + Dado.getInstance().getUltimoResultado());
            
            System.out.println("\n" + monotu.getCasilla(pos_actual).getNombre());
            
            pos_actual = monotu.nuevaPosicion(pos_actual, Dado.getInstance().tirar());
            
            cont ++;    
   
        }
               
        System.out.println("\n" + monotu.getPorSalida());
    }
}
